scene/gui/base_button.cpp

```diff
@@ -174,12 +176,6 @@ void BaseButton::on_action_event(Ref<InputEvent> p_event) {
 	}
 
 	if (!p_event->is_pressed()) {
-		Ref<InputEventMouseButton> mouse_button = p_event;
-		if (mouse_button.is_valid()) {
-			if (!has_point(mouse_button->get_position())) {
-				status.hovering = false;
-			}
-		}
```

Comment:
I don't see, how this removal has an impact on the MRPs of the initial issue description and of this PR.

Also wouldn't this removal create for certain projects situations, where the hovering status is kept `true`, while it should be `false`?

Perhaps this change could be investigated in a separate PR. The other two modifications are sufficient to resolve this problem in the issue description.

Reply:
See https://github.com/godotengine/godot/pull/93500#discussion_r1713121040
This is because when clicking on a child with mouse filter pass, this can make the hovering state incorrect and then the button can no longer be clicked the second time.
So I think this should still be removed for this PR.

- I have also tested to make sure that https://github.com/godotengine/godot/issues/33553 doesn't reappear with this removed.

The appearance when pressed and hovering over child is also incorrect, but this can be looked into later.

Reply:
I had forgotten about the MOUSE_FILTER_PASS case, I'll re-remove this.

Reply:
Thanks for the explanation. I agree.